src/libsystemd-network/sd-dhcp-server.c

```diff
@@ -633,8 +639,7 @@ static int parse_request(uint8_t code, uint8_t len, const void *option, void *us
                 break;
         case SD_DHCP_OPTION_MAXIMUM_MESSAGE_SIZE:
                 if (len == 2)
-                        req->max_optlen = be16toh(*(be16_t*)option) -
-                                          - sizeof(DHCPPacket);
+                        req->max_optlen = unaligned_read_be16(option) + sizeof(DHCPPacket);
```

Comment:
yeah, the code looks very much borked as it was previously. There this looks as if the "-" as accidentally duplicated and should just be one, and that a validity check to verify it's larger than sizeof(DHCPPacket) is missing.

@teg, could you comment, please?